doc/architecture/software-update.rst

```diff
@@ -21,11 +21,39 @@ Ostro OS based software can be deployed to a target device in two ways:
   component.
 
 
+Software Update
+===============
+The main concept behind the Software Update tool_ is that it doesn't
```

Comment:
Could we make the link name to be 'Software Update tool' instead of just 'tool' (use backticks for to enclose the complete text in this case)